### ISSUE TITLE
Search for folders checkbox in Find file dialog not synchronized with Hex control

### ISSUE THREAD

**Issue by rohitab** [FarGroup/FarManager#601]
### Far Manager version

3.0.6088.0 x64

### OS version

10.0.22623.1095

### Steps to reproduce

1. Go to a directory containing files and folders
2. Press `Alt+F7` to open _Find file_ dialog
3. If _Search for folders_ is not checked, press `Alt+o` to check it
4. Press `Enter` to search, then `Esc` to close the results
5. This is to ensure that `System.FindFolders` is set to `true`
6. Press `Alt+F7` to reopen _Find file_ dialog
7. Press `Alt+T` to switch to _Text_ search
8. Type in anything, say `Test`
9. Note that _Search for folders_ is not checked
10. Press `Alt+H` to switch to _Hex_ search
11. Erase the contents of the search box

### Expected behavior

_Search for folders_ should be checked, since the search box is empty.

### Actual behavior

_Search for folders_ is not checked, even though the search box is empty. If you now press `Enter` to search, the results will include folders, even though _Search for folders_ was not checked. If you reopen the _Find file_ dialog, _Search for folders_ will now be checked.

Note that if you manually check or uncheck _Search for folders_, the control will no longer check/uncheck based on whether the search box is empty or not. 

If you uncheck _Search for folders_ and press `Enter` to search, the next time you reopen the _Find file_ dialog, the control will not automatically check/uncheck based on the state of the search box. However, if you check _Search for folders_, and press `Enter` to search, the next time you reopen the _Find file_ dialog, the control will automatically check/uncheck based on the state of the search box.

To summarize, sometimes the _Search for folders_ automatically checks/unchecks, and other times it does not. It depends on the previous state and whether you have manually changed the state during the current dialog instance.

Wouldn't it be better to enable/disable _Search for folders_ instead of checking/unchecking it? That way, whatever state the user left it in, would not change automatically. This would be consistent with the other checkbox controls in the dialog.

**Comment by MKadaner** [comment]
The `Search for folders` was enabled / disabled in 6041. I broke it when I rearranged Text / Hex controls on Find File dialog. Looking... Thank you for reporting.

**Comment by alabuzhev** [comment]
> The `Search for folders` was enabled / disabled in 6041. Apparently, I broke it when I rearranged Text / Hex

I removed enabling / disabling the checkbox in 487c36d.
Conceptually switching between text / hex should not affect the find folders checkbox in any way: folders search does not depend on *how* we search for content, but whether or not we search for content in the first place. In other words, it makes sense to disable it if the "containing" field is not empty.

**Comment by alabuzhev** [comment]
However, all this blocking/unblocking and checking/unchecking doesn't make much sense in the first place:
- When "containing" is empty, the search predicate is basically "a file must match the mask"
- When "containing" is not empty, the search predicate is basically "a file must match the mask and have the specified data". Folders don't have a data stream by definition, so they will not be found regardless of the checkbox state, so the whole thing is purely cosmetic.

After 6090 the checkbox state is not affected by anything.

**Comment by rohitab** [comment]
Thanks for the update. The old implementation was a bit confusing. I like the way it's done now.

Tested on version `3.0.6090.0 x64`. It looks good.
